src/servers/src/mysql/handler.rs

```
@@ -160,6 +163,19 @@ impl MysqlInstanceShim {
     /// Save query and logical plan with a given statement key
     fn save_plan(&self, plan: SqlPlan, stmt_key: String) {
         let mut prepared_stmts = self.prepared_stmts.write();
+        let max_capacity = self.prepared_max_capacity;
+
+        let is_update = prepared_stmts.contains_key(&stmt_key);
+
+        if !is_update && prepared_stmts.len() >= max_capacity {
+            // Log a warning but don't insert
```

We must return an error instead of just printing an warning log.